src/components/Header/AppHeaderUser.tsx

```diff
@@ -74,7 +69,7 @@ export function AppHeaderUser({
 
         if (user) {
           // User exists, check if email_address is present
-          if (user.email_address) {
+          if (user.email_address && user.email_address !== "") {
```

Comment:
Empty string is falsy, so this may be rewritten: `if (user?.email_address) {` with optionals support.